livekit-plugins/livekit-plugins-assemblyai/livekit/plugins/assemblyai/stt.py

```diff
@@ -245,15 +288,53 @@ async def recv_task():
                 except Exception:
                     logger.exception("failed to process AssemblyAI message")
 
-        tasks = [
-            asyncio.create_task(send_task()),
-            asyncio.create_task(recv_task()),
-        ]
+        async def _wait_for_reconnect():
+            await self._reconnect_event.wait()
+
+        while True:
+            try:
+                ws = await self._connect_ws()
+                tasks = [
+                    asyncio.create_task(send_task(ws)),
+                    asyncio.create_task(recv_task(ws)),
+                ]
+                reconnect_task = asyncio.create_task(_wait_for_reconnect())
+
+                try:
+                    done, _ = await asyncio.wait(
+                        [asyncio.gather(*tasks), reconnect_task],
+                        return_when=asyncio.FIRST_COMPLETED,
+                    )
+                    if reconnect_task in done and self._reconnect_event.is_set():
+                        self._reconnect_event.clear()
+                        continue
+                    else:
+                        break
```

Comment:
Simpler to read this way
```suggestion
                    if reconnect_task not in done:
                        break
                        
                    self._reconnect_event.clear()
```